package/config.js

```diff
@@ -20,6 +20,14 @@ if (isArray(app.extensions) && app.extensions.length) delete defaults.extensions
 
 const config = deepMerge(defaults, app)
 config.outputPath = resolve('public', config.public_output_path)
-config.publicPath = `/${config.public_output_path}/`.replace(/([^:]\/)\/+/g, '$1')
+config.publicPath = `/${config.public_output_path}/`
+
+// Add prefix to publicPath.
+if (process.env.RAILS_RELATIVE_URL_ROOT) {
+  config.publicPath = `/${process.env.RAILS_RELATIVE_URL_ROOT}${config.publicPath}`
+}
+
+// Remove extra slashes.
+config.publicPath = config.publicPath.replace(/(^\/|[^:]\/)\/+/g, '$1')
```

Comment:
The `[^:]\/` part doesn't seem correct because if `public_output_path` contains a protocol, then `publicPath` would have a leading slash; e.g., (ignoring relative root) `public_output_path='https://cdn.com'` and `publicPath='/https://cdn.com/'`. The protocol would be preserved with the current RegExp, but that doesn't appear to be correct. Maybe I'm missing something.

Reply:
Yeah, but at the moment Rails handles the CDN side of things so it's fine but you are right this needs more check if we expect the `public_output_path` to have a protocol as well (which we don't at the moment).

Reply:
Sounds good.